### PR TITLE
added ncore site

### PR DESCRIPTION
**Pull request by DDanii**
### Config usage if relevant (new plugin or updated schema):
```
ncore:
  username: 'username'
  password: 'password'
  category: ['hd_hun', 'xvid_hun'] #optional
  sort_by: 'size' #optional
  sort_reverse: true #optional
```
### Log and/or tests output (preferably both):
```
2022-06-06 22:30:24 VERBOSE  discover      movies search   Discovering 6 titles ...
2022-06-06 22:30:24 INFO     discover      movies search   Ignoring interval because of --discover-now
2022-06-06 22:30:24 VERBOSE  discover      movies search   Searching for `Spider-Man: No Way Home (2021)` with plugin `ncore` (1 of 6)
2022-06-06 22:30:25 VERBOSE  discover      movies search   Searching for `Inception (2010)` with plugin `ncore` (2 of 6)
2022-06-06 22:30:26 VERBOSE  discover      movies search   Searching for `The Lost City (2022)` with plugin `ncore` (3 of 6)
2022-06-06 22:30:27 VERBOSE  discover      movies search   Searching for `Fantastic Beasts: The Secrets of Dumbledore (2022)` with plugin `ncore` (4 of 6)
2022-06-06 22:30:28 VERBOSE  discover      movies search   Searching for `Sonic the Hedgehog 2 (2022)` with plugin `ncore` (5 of 6)
2022-06-06 22:30:29 VERBOSE  discover      movies search   Searching for `The Matrix (1999)` with plugin `ncore` (6 of 6)
2022-06-06 22:30:30 VERBOSE  details       movies search   Produced 70 entries.
```
#### To Do:

-

### CODE REVIEW

**Comment by BrutuZ** [comment]
Forwarding a couple comments I received on previous contributions:

- Any particular reason to have the requests data formatted as a string instead of a dict?
- Could turn the plugin into a generator (use `yield` inside of the loop instead of `return` after) to make it compatible with the `limit` plugin.

**Comment by DDanii** [comment]
Thanks For the comments.

- That is the format the browser sent to the site. I'm not sure how I would generate that from a dict.
- Changed it to yield.

**Comment by BrutuZ** [comment]
> I'm not sure how I would generate that from a dict

You don't have to, Requests accepts dicts for `data` and handles them accordingly

**Comment by DDanii** [comment]
Thanks for the heads up

File: flexget/components/sites/sites/ncore.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            soup = get_soup(page.content)
+            for a in soup.findAll('a', title=re.compile(".+")):
+                if 'details' in a.get('href'):
+                    e = Entry()
+
+                    href = a.get('href')
+                    id = href[href.find("id=") + 3 :]
+
+                    e['title'] = a.text
+                    e['url'] = URL + '/torrents.php?action=download&id={0}&'.format(id) + PASSKEY
```
Comment (paranoidi) [inline flexget/components/sites/sites/ncore.py:114]: f-strings are nice, using urllib is the best though

but this is probably fine for this PR if you don't feel like changing it

File: flexget/components/sites/sites/ncore.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        soup = get_soup(page.content)
+        passkeyLine = str(soup.find('link', href=re.compile(r'rss\.php\?key=')))
+        PASSKEY = passkeyLine[passkeyLine.find("key=") : passkeyLine.find('"', 20, 90)]
+
+        entries = set()
+
+        for search_string in entry.get('search_strings', [entry['title']]):
+            data = {"mire": search_string, "miben": "name", "miszerint": SORT[config['sort_by']]}
+
+            if len(config['category']) > 0:
```
[the remark this answers is not in the document]
Comment (DDanii) [reply flexget/components/sites/sites/ncore.py:95]: If I remove the len:
TypeError: '>' not supported between instances of 'list' and 'int'

[the remark this answers is not in the document]
Comment (BrutuZ) [reply flexget/components/sites/sites/ncore.py:95]: He means you can test it simply with
```python
if config['category']:
```
Probably best to go for `config.get('category')` though, that won't cause an exception if the category key doesn't exist

[the remark this answers is not in the document]
Comment (DDanii) [reply flexget/components/sites/sites/ncore.py]: Thanks

**Comment by DDanii** [comment]
@paranoidi It is good now?

**Comment by paranoidi** [comment]
Yep, thanks, merged!

**Pull request merged**